File: programmers/greedy/체육복.py
def solution(n, lost, reserve):
    reserve.sort()
    lost.sort()
    while reserve and lost:
        for r in reserve:
            if r in lost:
                lost.remove(r)
                reserve.remove(r)
                break
        else:
            r = reserve.pop(0)
            for l in lost:
                if r - 1 == l:
                    lost.remove(l)
                    break
                elif r + 1 == l:
                    lost.remove(l)
                    break
    else:
        return n - len(lost)
   
if __name__ == "__main__":
    # print(solution(6, [3,4,5], [3,4,6]))
    # print(solution(4, [2,3], [3,4]))
    # print(solution(4, [1, 4], [2, 3] ))
    # print(solution(4, [2], [1, 2] ))
    # print(solution(5, [3, 4], [4, 3]))
    print(solution(5, [4,5], [3, 4]))

File: programmers/코딩테스트연습/그리디/체육복.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(int n, vector<int> lost, vector<int> reserve) {
    vector<int> realLost;
    for(int i = 0; i < lost.size(); i++){
        int lostNumber = lost[i];
        if(find(reserve.begin(), reserve.end(), lostNumber) == reserve.end()){
            realLost.push_back(lost[i]);
        } else{
            reserve.erase(find(reserve.begin(), reserve.end(), lostNumber));
        }
    }
    sort(realLost.begin(), realLost.end());
    sort(reserve.begin(), reserve.end());
    int count = 0;
    for(int i = 0; i < realLost.size(); i++){
        int lostNumber = realLost[i];
        auto it = find_if(reserve.begin(), reserve.end(), [lostNumber](int r){
            return lostNumber - 1 == r || lostNumber + 1 == r;
        });
        if(it == reserve.end()){
            count++;
        } else{
            reserve.erase(it);
        }
    }
    return n - count;
}